pkg/api/vlabs/types.go

```diff
@@ -118,6 +118,14 @@ const (
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
 	OrchestratorType OrchestratorType `json:"orchestratorType"`
+	KubernetesConfig KubernetesConfig `json:"kubernetesConfig,omitempty"`
+}
+
+// KubernetesConfig contains the Kubernetes config structure, containing
+// Kubernetes specific configuration
+type KubernetesConfig struct {
```

Comment:
I prefer the flat one. Introduce more configurations or concepts like "sovereignconfig" may cause more learning effort for users. @amanohar @colemickens what do you think about it?